pycbc/events/stat.py

```diff
@@ -1741,6 +1755,39 @@ def logsignalrate(self, stats, shift, to_shift):
         logr_s += numpy.log((self.curr_mchirp / 20.) ** (11. / 3.))
         return logr_s
 
+    def single(self, trigs):
+        """
+        Calculate the necessary single detector information
+
+        In this case the ranking rescaled (see the lognoiserate method here)
+        with the phase, end time, sigma, SNR, template_id and the
+        benchmark_logvol values added in. This also stored the current chirp
+        mass for use when computing the coinc statistic values.
+
+        Parameters
+        ----------
+        trigs: dict of numpy.ndarrays, h5py group or similar dict-like object
+            Object holding single detector trigger information.
+
+        Returns
+        -------
+        numpy.ndarray
+            The array of single detector values
+        """
+        from pycbc.conversions import mchirp_from_mass1_mass2
+        try:
+            mass1 = trigs.param['mass1']
+            mass2 = trigs.param['mass2']
+        except AttributeError:
+            mass1 = trigs['mass1']
+            mass2 = trigs['mass2']
+        self.curr_mchirp = mchirp_from_mass1_mass2(mass1, mass2)
+
+        if self.mcm is not None:
+            # Careful - input might be a str, so cast to float
+            self.curr_mchirp = min(self.curr_mchirp, float(self.mcm))
```

Comment:
I should have noticed this earlier, but `self.curr_mchirp` is an array, right? How does `min()` work here?

Reply:
Like the above, this is not new. I don't quite understand how the `min()` works here, but Arthur said in the previous review thread `self.mcm` is None in PyCBC Live, and the offline case is unchanged.

Reply:
I'm confused why this works as well - it shouldn't.

I think this should just use np.minimum instead, but I'm not sure if we ever depend on the fact that ReadByTemplate has the same mchirp for all triggers when this is called

Reply:
Since this is not a change that Arthur made (its being shown as a change due to Arthur moving code around) I support just leaving it as-is.

Reply:
I think Arthur has introduced a use-case for it though (mchirp kwargs being passed through to rank_stat_coinc)

I will have a chat with Arthur and see what we can do to solve this, but it will be next week

Reply:
Arthur said here: https://github.com/gwastro/pycbc/pull/4527/files/640a68b622f1267a82960ca4d56f51b214f8ff94#r1476198713 that self.mcm will be None in PyCBC Live

Reply:
True, it wont be used anytime soon I don't think

Reply:
Between that statement and the lack of any actual code change, I wouldn't let this bit of code delay the review as a whole